### DISCUSSION TITLE
Reset dimmer after .ui.dimmer removed

### DISCUSSION THREAD

**Discussion by justinhunt1223** [fomantic/Fomantic-UI#1314] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
I migrated from Semantic UI for a few updates and am trying to solve a problem. If a segment is dimmed, then the contents of that segment are changed (such as from an ajax call) and the .ui.dimmer child element is deleted, the dimming on that segment seems not work anymore and things go crazy. I'm looking for a way to reset the dimming capabilities on the segment.

[jsfiddle](https://jsfiddle.net/0hoLadbr/1/)

I've gotten around this by never touching the .ui.dimmer and adding a child to the .segment that is the content but I feel like there should be a way to reset the dimmer.

**Comment by lubber-de** [comment, marked as answer]
This is already possible by calling the `preinitialize` behavior prior to hiding the dimmer.
See https://jsfiddle.net/usje9cyz/

**Comment by justinhunt1223** [comment]
Awesome, that's exactly what I was looking for.

**Comment by lubber-de** [comment]
>When I click on the button, the dimmer shows and then hides correctly, but clicking the button again does not show the dimmer again. Was it supposed to?

That's because your example is overwriting the html and then the button trigger is not available anymore ;)

**Comment by justinhunt1223** [comment]
I noticed that right after I posted it. You were too quick!
